### ISSUE TITLE
[arrow-parens] separate options for async functions

### ISSUE THREAD

**Issue by ab-pm** [eslint/eslint#12439]
**What rule do you want to change?**

`arrow-parens` (currently on eslint 6.5.1).

**Does this change cause the rule to produce more or fewer warnings?**

fewer warnings.

**How will the change be implemented?**

New option / different option layout.

**Please provide some example code that this change will affect:**

```js
withCallback(x => { … });
withCallback(async (x) => { … });
```
```json
…
	'arrow-parens': ["error", "as-needed"],
// or
	'arrow-parens': ["error", "always"],
```

**What does the rule currently do for this code?**

It either accepts a version with parenthesis on both functions or a version without on both functions, but no mixed style. I would prefer my `async` arrow functions to always have parenthesis while my normal arrow functions should have them only when needed.

**What will the rule do after it's changed?**

It should be able to lint and fix parens on normal arrow functions and async arrow functions separately.

**Are you willing to submit a pull request to implement this change?**

Yes, but I don't know how the configuration option should look like to be flexible, separable by function type, and backwards-compatible. Can you suggest something?

**Comment by kaicataldo** [comment]
@ab-pm This generally sounds like a reasonable request, but we ask that you please feel out the issue template as requested. Thank you!

**Comment by ab-pm** [comment]
OK, updated, I thought the template was more of a guidance I had hoped to include all relevant information to start a discussion in a more concise format.

**Comment by kaicataldo** [comment]
Thanks for filling that out! I'm in support of this option, pending deciding on finalizing the API.
